### ISSUE TITLE
Crash with EXC_BAD_ACCESS using map.queryItems right after opening a map

### ISSUE THREAD

**Issue by sammyfdm** [twilio/twilio-sync-ios#3]
I am writing to a TWSDocument for the first time using the iOS SDK and came across a crash whenever I left out the completion handler from setData or mutateData calls. Here's an example of the code I used that would crash:
```
DispatchQueue(label: "app_state_document_queue").sync { [weak self] in
    appStateDocument?.mutateData({ existingData in
        existingData?[key] = value
        return existingData
    }, metadata: nil)
}
```
...and it stopped causing the crash as soon as I added a completion handler, even if all it did was log:
```
DispatchQueue(label: "app_state_document_queue").sync { [weak self] in
    appStateDocument?.mutateData({ existingData in
        existingData?[key] = value
        return existingData
    }, metadata: nil, completion: { result, updatedData in
        self?.logDebug("Updated app state document \(key) to \(value): \(updatedData)")
    })
}
```

The SDK code is obfuscated, but the stack trace shows it throwing a `EXC_BAD_ACCESS` in `[TWSDocument fulfillForDataCallbackWithCompletion]`. I suspect there's a nil check missing or something...

**Comment by sammyfdm** [comment]
Apologies, I was on an old version of the TwilioSync SDK (`0.8.7`), I have updated to `1.2.1` and still get a crash, but this one looks like it's related to maps, not the completion handler for Documents.:

```
#1	0x0000000107148dc4 in invocation function for block in +[TWSHelpers processMapPaginatorWithCompletion:delegateWrapper:]::$_2::operator()(rtd::TSOperationResult<std::__1::shared_ptr<rtd::ITSPaginator<std::__1::shared_ptr<rtd::sync::MapItem const> > > >) const at /Users/distiller/rtd-sdk-monorepo/sdk/ios/sync/TwilioSyncClient/Sources/TWSHelpers.mm:125
```

**Comment by sammyfdm** [comment]
Some additional digging leads me to believe the culprit is calling the `map.queryItems(with: TWSMapQueryOptions, completion: TWSMapPaginatorCompletion)` function. I have also tried using the async version of the method with the same results, and have tried providing different options, but it always crashees with an EXC_BAD_ACCESS in the same way.

**Comment by sammyfdm** [comment]
I figured out that there is a work around by delaying the query. Here's my code that has been working through some tests I've run on it:

```
func onMapOpened(_ map: TWSMap) {
    DispatchQueue(label: "delayed_map_initialization").asyncAfter(
        deadline: DispatchTime.now() + 2,
        execute: DispatchWorkItem(block: { [weak self] in
            participantMap.queryItems(with: TWSMapQueryOptions(), completion: { (result, paginator) in
                ...
            })
        })
    )
}
```

**Comment by berkus** [comment]
Thanks for the report! Looks like this definitely shouldn't be happening. We'll look into it.

**Comment by Gray-Wind** [comment]
We have just released v2.0, could you check the reproduction on the latest version please?

**Comment by Gray-Wind** [comment]
We've identified the issue and the next version will have the fix.

**Comment by sammyfdm** [comment]
I don't seem to be getting the crash anymore in v2.0. Did you identify an issue with v2.0 related to this?

**Comment by Gray-Wind** [comment]
Yes, we've identified an issue with the similar to provided by you code on v2.0. Long story short, it is better for now to use an object returned by SyncClient itself and not to use an object returned with `onOpened...` delegate method.

**Comment by Gray-Wind** [comment]
Hey @sammyfdm,

We've just released iOS SDK version 2.0.1 with the fix included.

I am closing this ticket, feel free to open a new one.
